enhancements/monitoring/user-workload-monitoring.md

```diff
@@ -291,6 +313,79 @@ remapping the values of labels as we base our tenancy model on this. We have two
+
+#### `monitoring-rules-edit`
+This is a cluster role which can be bound against a concrete namespace by the cluster admin.
+Embeds the `create/edit/delete prometheusrules.monitoring.coreos.com` permission. It allows to:
+
+1. create/modify/delete PrometheusRule custom resources matching the permitted namespace.
+
+```
+$ oc -n <namespace> create/patch/delete prometheusrule <foo>.
+```
+
+2. Silence firing alerts from the Alertmanager `/alerts?namespace=<foo>` endpoint matching the permitted namespace.
+
+#### `monitoring-targets-edit`
+This is a cluster role which can be bound against a concrete namespace by the cluster admin.
+Embeds the `get/create/edit/delete` permissions for the following custom resources:
+
+1. ServiceMonitor
+2. PodMonitor
+3. PrometheusRule
+
+It allows to create new scraping targets for services/pods and allows to create new recording or alerting rules.
+
+#### `monitoring-workload-config-edit`
+This is a named role against the `workload-monitoring-config` configmap resource in the `openshift-monitoring` namespace
+embedding `get/create/edit/delete` permissions.
+
+### Configuration/Enablement
+
+User workload monitoring is not enabled by default. It must be explicitely enabled by the user.
+User workload monitoring is enabled just by creating an empty `workload-monitoring-config` configmap in the `openshift-monitoring` namespace.
+
+The user workload monitoring configuration allows to set all existing configuration options as for the in-cluster stack with the addition to set the replica count of the user workload monitoring Prometheus instances.
+
+Note: Long term the existing `cluster-monitoring-config` and `workload-monitoring-config` will be moved to dedicated custom resources and CRDs but are out of scope for this enhancement.
+
+### Resource impact
+
+Once user workload monitoring is enabled, by default 2 replicas of Prometheus and one Prometheus Operator instance are being deployed in the `openshift-user-workload-monitoring` namespace.
+
+As of OpenShift 4.4 the following idle resource usage have been measured after 24h of staying idle:
+
+1. Prometheus: ~65MB memory usage per replica instance, ~6 milli-cores CPU usage
+2. Prometheus Operator: ~15MB memory usage, ~1 milli-core of CPU usage
+
+This results in a total overhead of ~145MB memory and ~7 milli-cores of CPU usage. The above resource measurements will be set as the default resource requests.
+
+### High cardinality detection
+
+As a first step a similar high cardinality alert will be used as in [9] and potentially [10]. More formally we will alert on:
+
+- A sum over time of scrape samples scraped reaching a threshold.
+- A sum over time of scrape series being added reaching a threshold.
+- A sum of the rate of tsdb head series created reaching a threshold.
+
+These alerts will fire in `info` level only and transmitted via Telemetry.
+Over time we will learn about better thresholds and potentially also about better alerts.
```

Comment:
Can you clarify how we can achieve this if we only send the firing alerts. I think we will need to send more metrics via telemetry about user workload usage, rather than just the firing alerts.

Reply:
which ones do you suggest?

Reply:
wait, we have a couple already defined in https://issues.redhat.com/browse/MON-968. we could add them here.

Reply:
Yes those sound good, just find this a bit misleading to suggest we can use the alert only to modify the thresholds, without the extra info we would be sending. e.g. we want to mention that we to send extra data via telemetry for user workload and cluster monitoring in the KEP, wdyt?

Reply:
agreed, i will add some language from the referenced jira.

Reply:
added metrics from https://issues.redhat.com/browse/MON-968